### ISSUE TITLE
HomeWhiz Component blocking Homeassistant shutdown

### ISSUE THREAD

**Issue by TechHummel** [home-assistant-HomeWhiz/home-assistant-HomeWhiz#137]
```
2023-07-06 22:46:16.901 WARNING (MainThread) [homeassistant.core] Timed out waiting for shutdown stage 3 to complete, the shutdown will continue
2023-07-06 22:46:16.901 WARNING (MainThread) [homeassistant.core] Shutdown stage 3: still running: <Task pending name='Task-20125' coro=<HomewhizBluetoothUpdateCoordinator.try_reconnect() running at /config/custom_components/homewhiz/bluetooth.py:77> wait_for=<Future pending cb=[Task.task_wakeup()]> cb=[set.remove()]>
2023-07-06 22:46:16.902 WARNING (MainThread) [homeassistant.core] Shutdown stage 3: still running: <Task pending name='Task-20112' coro=<HomewhizBluetoothUpdateCoordinator.try_reconnect() running at /config/custom_components/homewhiz/bluetooth.py:74> wait_for=<Future pending cb=[Task.task_wakeup()]> cb=[set.remove()]>
```
Seems like our custom component is sometimes blocking shutdown.

Maybe in bluetooth.py
```py
    async def try_reconnect(self) -> None:
        _LOGGER.debug(f"[{self.address}] Trying to reconnect")
        while self.alive and not self.is_connected:
            if not bluetooth.async_address_present(
                self.hass, self.address, connectable=True
            ):
                _LOGGER.info(
                    "Device not found. "
                    "Will reconnect automatically when the device becomes available"
                )
                return
            try:
                _LOGGER.debug(f"[{self.address}] Establish connection from reconnect")
                await self.connect()
            except Exception:
                _LOGGER.info("Can't reconnect. Waiting a minute to try again")
                await asyncio.sleep(60)
```

needs to be changed to
```py
    async def try_reconnect(self) -> None:
        _LOGGER.debug(f"[{self.address}] Trying to reconnect")
        if self.alive and not self.is_connected:
            if not bluetooth.async_address_present(
                self.hass, self.address, connectable=True
            ):
                _LOGGER.info(
                    "Device not found. "
                    "Will reconnect automatically when the device becomes available"
                )
                return
            try:
                _LOGGER.debug(f"[{self.address}] Establish connection from reconnect")
                await self.connect()
            except Exception:
                _LOGGER.info("Can't reconnect. Waiting a minute to try again")
                await asyncio.sleep(60)
                self.hass.create_task(self.try_reconnect())
```

**Comment by TechHummel** [comment]
Maybe this got fixed on Homeassistant side: https://github.com/home-assistant/core/pull/104150

**Comment by TechHummel** [comment]
Closed for now, maybe fixed with #200
